neqo-qpack/src/qlog.rs

```diff
@@ -0,0 +1,60 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Functions that handle capturing QLOG traces.
+
+use crate::Res;
+use neqo_common::qlog::NeqoQlogRef;
+use qlog::{event::Event, QPackInstruction, QpackInstructionTypeName};
+use std::fmt::LowerHex;
+
+// TODO(hawkinsw@obs.cr): There is a copy of this in neqo-transports/src/qlog.rs.
+// Refactor both uses into something in neqo-common.
+fn slice_to_hex_string<T: LowerHex>(slice: &[T]) -> String {
+    if slice.is_empty() {
+        "0x0".to_string()
+    } else {
+        slice
+            .iter()
+            .fold("0x".to_string(), |acc, x| acc + &format!("{:x}", x))
```

Comment:
See below for the bug.

Reply:
Is the bug to which you refer the one about the format string of {:02x} vs {:0x}? "below" and "above" don't seem to be tracking when I look at this version. Sorry!

Reply:
@martinthomson  I know that @agrover changed this to using `hex()` but I would still really know the "bug" to which you are referring. @martinthomson indicated that it is "clearly wrong" but I can't seem to find the "clearly wrong" except for the format string padding problem.